hooks/src/lib/utils/getCardBrand.ts

```diff
@@ -0,0 +1,47 @@
+import { CARD_BRAND } from "../constants/option";
+
+const getCardBrand = (cardNumber: string) => {
+  if (checkIsVisa(cardNumber)) return CARD_BRAND.Visa;
+  if (checkIsMaster(cardNumber)) return CARD_BRAND.Master;
+  if (checkIsDiners(cardNumber)) return CARD_BRAND.Diners;
+  if (checkAMEX(cardNumber)) return CARD_BRAND.AMEX;
+  if (checkIsUnionPay(cardNumber)) return CARD_BRAND.UnionPay;
+
+  return CARD_BRAND.Normal;
+};
+
+const checkIsUnionPay = (cardNumber: string) => {
+  const firstThreeDigits = cardNumber.slice(0, 3);
+  const firstFourDigits = cardNumber.slice(0, 4);
+  const firstSixDigits = cardNumber.slice(0, 6);
+
+  if ("624" <= firstThreeDigits && firstThreeDigits <= "626") return true;
```

Comment:
`string <= string` 구문에서 일어나는 암시적인 형변환을 `number <= number` 형태로 바꾸면 어떨까요?

Reply:
```suggestion
  if (624 <= Number(firstThreeDigits) && Number(firstThreeDigits) <= 626) return true;
```

요렇게 바꿀 수 있을거 같은데 string 으로도 비교가 가능한 연산을 number 타입으로 바꾸어 계산하는게 어떠한 이점이 있을까요??

제가 생각하기에 직접 number 타입으로 바꿔서 연산했을 때 이점은 `혹시 모를 타입 안정성 (string 비교 연산이 예상대로 흘러가지 않는경우)` 밖에 생각나지 않아요..!

아니면 가독성 측면에서 Number 로 타입 형변환을 하는게 더 좋을까요??

Reply:
암시적 형변환을 명시적으로 변경하는 것은 가독성 측면으로 봐주시면 맞을 것 같습니다 :)